raiden/network/proxies/token_network.py

```diff
@@ -0,0 +1,773 @@
+                amount_to_deposit,
+            ))
+
+        if current_balance < amount_to_deposit:
+            raise ValueError('deposit [{}] cant be larger than the available balance [{}].'.format(
+                amount_to_deposit,
+                current_balance,
+            ))
+
+        log.info(
+            'deposit called',
+            token_network=pex(self.address),
+            node=pex(self.node_address),
+            partner=pex(partner),
+            total_deposit=total_deposit,
+            amount_to_deposit=amount_to_deposit,
+        )
+
+        if not self.channel_operations_lock.acquire(blocking=False):
+            raise ChannelBusyError(
+                f'Channel with address {self.address} is '
+                f'busy with another ongoing operation.'
+            )
+
+        with releasing(self.channel_operations_lock):
+            transaction_hash = estimate_and_transact(
+                self.proxy,
+                'setTotalDeposit',
+                self.node_address,
+                total_deposit,
+                partner,
+            )
+
+            self.client.poll(
+                unhexlify(transaction_hash),
+                timeout=self.poll_timeout,
+            )
+
+            receipt_or_none = check_transaction_threw(self.client, transaction_hash)
+            if receipt_or_none:
+                log.critical(
+                    'deposit failed',
+                    token_network=pex(self.address),
+                    node=pex(self.node_address),
+                    partner=pex(partner),
+                    total_deposit=total_deposit,
+                )
+
+                self._check_exists()
+                raise TransactionThrew('Deposit', receipt_or_none)
+
+            log.info(
+                'deposit successful',
+                token_network=pex(self.address),
+                node=pex(self.node_address),
+                partner=pex(partner),
+                total_deposit=total_deposit,
+            )
+
+    def close(
+            self,
+            partner: typing.Address,
+            nonce: typing.Nonce,
+            balance_hash: typing.BalanceHash,
+            additional_hash: typing.AdditionalHash,
+            signature: typing.Signature
+    ):
+        """ Close the channel using the provided balance proof.
+
+        Raises:
+            ChannelBusyError: If the channel is busy with another operation.
+        """
+
+        log.info(
+            'close called',
+            token_network=pex(self.address),
+            node=pex(self.node_address),
+            partner=pex(partner),
+            nonce=nonce,
+            balance_hash=encode_hex(balance_hash),
+            additional_hash=encode_hex(additional_hash),
+            signature=encode_hex(signature),
+        )
+
+        if not self.channel_operations_lock.acquire(blocking=False):
+            raise ChannelBusyError(
+                f'Channel with address {self.address} is '
```

Comment:
This won't work. Address should be given with `pex()` but also we no longer have channel addresses. Use the channel identifier instead.